File: src/main/java/com/robohorse/robopojogenerator/generator/processors/InnerObjectResolver.java
package com.robohorse.robopojogenerator.generator.processors;

import com.robohorse.robopojogenerator.generator.consts.ClassType;
import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Created by vadim on 23.09.16.
 */
public abstract class InnerObjectResolver {

    public void resolveClassType(Object object) {
        if (object instanceof JSONObject) {
            onJsonObjectIdentified();

        } else if (object instanceof JSONArray) {
            onJsonArrayIdentified();

        } else if (object instanceof String) {
            onInnerObjectIdentified(ClassType.STRING);

        } else if (object instanceof Integer) {
            onInnerObjectIdentified(ClassType.INTEGER);

        } else if (object instanceof Double) {
            onInnerObjectIdentified(ClassType.DOUBLE);

        } else if (object instanceof Float) {
            onInnerObjectIdentified(ClassType.FLOAT);

        } else if (object instanceof Long) {
            onInnerObjectIdentified(ClassType.LONG);

        } else if (object instanceof Boolean) {
            onInnerObjectIdentified(ClassType.BOOLEAN);

        } else {
            onInnerObjectIdentified(ClassType.OBJECT);
        }
    }

    public abstract void onInnerObjectIdentified(ClassType classType);

    public abstract void onJsonObjectIdentified();

    public abstract void onJsonArrayIdentified();
}

File: generator/src/main/kotlin/com/robohorse/robopojogenerator/generator/processing/InnerObjectResolver.kt
package com.robohorse.robopojogenerator.generator.processing

import com.robohorse.robopojogenerator.generator.common.ClassEnum
import com.robohorse.robopojogenerator.generator.common.ClassEnum.BOOLEAN
import com.robohorse.robopojogenerator.generator.common.ClassEnum.DOUBLE
import com.robohorse.robopojogenerator.generator.common.ClassEnum.FLOAT
import com.robohorse.robopojogenerator.generator.common.ClassEnum.INTEGER
import com.robohorse.robopojogenerator.generator.common.ClassEnum.LONG
import com.robohorse.robopojogenerator.generator.common.ClassEnum.OBJECT
import com.robohorse.robopojogenerator.generator.common.ClassEnum.STRING
import org.json.JSONArray
import org.json.JSONObject

internal abstract class InnerObjectResolver {

    fun resolveClassType(targetItem: Any) {
        when (targetItem) {
            is JSONObject -> onJsonObjectIdentified()
            is JSONArray -> onJsonArrayIdentified()
            is String -> onInnerObjectIdentified(STRING)
            is Int -> onInnerObjectIdentified(INTEGER)
            is Double -> onInnerObjectIdentified(DOUBLE)
            is Float -> onInnerObjectIdentified(FLOAT)
            is Long -> onInnerObjectIdentified(LONG)
            is Boolean -> onInnerObjectIdentified(BOOLEAN)
            else -> onInnerObjectIdentified(OBJECT)
        }
    }

    abstract fun onInnerObjectIdentified(classEnum: ClassEnum?)

    abstract fun onJsonObjectIdentified()

    abstract fun onJsonArrayIdentified()
}
